v0.6.0 blog post (#929)
reflex-dev/reflex-web: docs/api-reference/var_system.md

    "Fail",
)
```

 The conditional to roughly the following in Javascript:

```js
state.threshold >= 50 ? "Pass" : "Fail";
```

## Overview

Simply put, a `Var` in Reflex represents a Javascript expression.
If the type is known, it can be any of the following:

- `NumberVar` represents an expression that evaluates to a Javascript `number`. `NumberVar` can support both integers and floating point values
- `BooleanVar` represents a boolean expression. For example: `false`, `3 > 2`.
- `StringVar` represents an expression that evaluates to a string. For example: `'hello'`, `(2).toString()`.
- `ArrayVar` represents an expression that evaluates to an array object. For example: `[1, 2, 3]`, `'words'.split()`.
- `ObjectVar` represents an expression that evaluates to an object. For example: `\{a: 2, b: 3}`, `\{deeply: \{nested: \{value: false}}}`.
- `NoneVar` represent null values. These can be either `undefined` or `null`.

## Creating Vars

State fields are converted to `Var` by default. Additionally, you can create a `Var` from Python values using `rx.Var.create()`:

```py
rx.Var.create(4) # NumberVar
rx.Var.create("hello") # StringVar
rx.Var.create([1, 2, 3]) # ArrayVar
```

> If you want to explicitly create a `Var` from a raw Javascript string, you can call the init function on `rx.Var`:

Reviewer: ```suggestion
If you want to explicitly create a `Var` from a raw Javascript string, you can instantiate `rx.Var` directly:
```

```py
rx.Var("2", _var_type=int).guess_type() # NumberVar
```

In the example above, `.guess_type()` will attempt to downcast from a generic `Var` type into `NumberVar`.
For this example, calling the function `.to(int)` can also be used in place of `.guess_type()`.

## Operations

The `Var` system also supports some other basic operations.

> For example, `NumberVar` supports basic arithmetic operations like `+` and `-` like in Python.

Reviewer: ```suggestion
For example, `NumberVar` supports basic arithmetic operations like `+` and `-`, as in Python.
```
